CHANGELOG.md

```diff
@@ -4,12 +4,21 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 ### Added
+- [#1103](https://github.com/plotly/dash/pull/1103) Wildcard IDs and callbacks. Component IDs can be dictionaries, and callbacks can reference patterns of components, using three different wildcards: `All`, `MATCH`, and `ALLSMALLER`, available from `dash.dependencies`. This lets you create components on demand, and have callbacks respond to any and all of them. To help with this, `dash.callback_context` gets three new entries: `outputs_list`, `inputs_list`, and `states_list`, which contain all the ids, properties, and except for the outputs, the property values from all matched components.
+
 - [#1134](https://github.com/plotly/dash/pull/1134) Allow `dash.run_server()` host and port parameters to be set with environment variables HOST & PORT, respectively
 
 ### Changed
+- [#1103](https://github.com/plotly/dash/pull/1103) Multiple changes to the callback pipeline:
+  - `dash.callback_context.triggered` now does NOT reflect any initial values, and DOES reflect EVERY value which has been changed either by activity in the app or as a result of a previous callback. That means that the initial call of a callback with no prerequisite callbacks will list nothing as triggering. For backward compatibility, we continue to provide a length-1 list for `triggered`, but its `id` and `property` are blank strings, and `bool(triggered)` is `False`.
+  - A callback which returns the same property value as was previously present will not trigger the component to re-render, nor trigger other callbacks using that property as an input.
```

Comment:
Is it important to distinguish between value-props and object/array-props here? Returning an object or array with the same props/values will trigger re-render.

Reply:
Actually turns out I didn't implement this for callback returns, only for front-end changes via `setProps`. And I used `R.equals` so no need to distinguish, it'll deep compare. I suppose that might be a problem for very large props (table data for example)... perhaps `===` would be safer? Might be good to do what I originally said I had done 😏 esp if we switch to `===`.

Anyway I fixed the changelog to match the current behavior in this branch in https://github.com/plotly/dash/pull/1103/commits/1e75e9b3fb17e37f261d0b7541a49f658c7fabc8